core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -474,6 +461,16 @@ private void commitSimpleTransaction() {
 
   private void cleanUpOnCommitFailure() {
     // the commit failed and no files were committed. clean up each update.
+    computeUncommittedFiles();
+
+    // delete all the uncommitted files
+    Tasks.foreach(deletedFiles)
+        .suppressFailureWhenFinished()
+        .onFailure((file, exc) -> LOG.warn("Failed to delete uncommitted file: {}", file, exc))
+        .run(ops.io()::deleteFile);
+  }
+
+  private void computeUncommittedFiles() {
```

Comment:
The diff looks a bit strange, but  it's a bit of refactoring which separates

1) computing the files to clean up via going through all the updates and calling `cleanAll` which enqueues the uncommitted files

2.) Actually deleting the files

I did this because in the code there a rereferences to "cleaning up" but really what's happening is an enqueuing of possible files to cleanup after the commit succeeds or fails.